pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -278,7 +278,7 @@ func (self *RefsHelper) CreateCheckoutMenu(ref string, branches []*models.Branch
 			LabelColumns: []string{branch.Name},
 			OnPress: func() error {
 				self.c.LogAction(self.c.Tr.Actions.CheckoutBranch)
-				return self.CheckoutRef(ref, types.CheckoutRefOptions{})
+				return self.c.Git().Branch.Checkout(branch.Name, git_commands.CheckoutOptions{})
```

Comment:
Why did you change this? I think `CheckoutRef` is the right thing to use here. It takes care of refreshing, and a bunch of other useful things that I think we want here too.

(Side note: I'm not a fan of "conventional commits"; seeing "fix:" in the subject line doesn't tell me anything except that this is supposed to fix something. I'd rather have a sentence in the commit message body that explains _what_ this fixes.)

Reply:
Because the code before had a bug and checked out the code as detached head.
Do I understand you correctly that passing the branch as ref argument should check out the branch and refresh the view?
I will have a look again.

Reply:
Sorry that was my fault. I was distracted testing the code, then fixed it based on my own assumptions and had forgotten, that you already provided the correct solution.

Reply:
Fixed with the last push.